src/util/options.rs

```diff
@@ -389,105 +399,199 @@ impl FromStr for AffinityKind {
     /// lower bounds. Note that this is considered less performant than a Bounded nursery since a
     /// Fixed nursery size can be too restrictive and cause more GCs.
-    Fixed,
-}
-
-#[derive(Copy, Clone, Debug)]
-/// An option that provides a min/max interface to MMTk and a Bounded/Fixed interface to the
-/// user/VM.
-pub struct NurserySize {
-    /// The nursery type
-    pub kind: NurseryKind,
-    /// Minimum nursery size (in bytes)
-    pub min: usize,
-    /// Maximum nursery size (in bytes)
-    max: Option<usize>,
+    Fixed(usize),
 }
 
 impl NurserySize {
-    /// Create a NurserySize with the given kind. The value argument specifies the nursery size
-    /// for a fixed nursery, or specifies the max size for a bounded nursery.
-    pub fn new(kind: NurseryKind, value: Option<usize>) -> Self {
-        match kind {
-            NurseryKind::Bounded => NurserySize {
-                kind,
-                min: DEFAULT_MIN_NURSERY,
-                max: value,
-            },
-            NurseryKind::Fixed => NurserySize {
-                kind,
-                min: value.unwrap(),
-                max: value,
-            },
-        }
+    /// Estimate the virtual memory needed for the nursery space. This is used during space creation.
+    pub fn estimate_virtual_memory_in_bytes(&self) -> usize {
+        let virtual_memory_bytes = match *self {
+            NurserySize::Bounded { min: _, max } => max,
+            // Just use the default max nursery size -- the nursery won't get larger than that.
+            NurserySize::ProportionalBounded { min: _, max: _ } => {
+                if !crate::util::heap::vm_layout::vm_layout().force_use_contiguous_spaces {
+                    DEFAULT_MAX_NURSERY_32
+                } else {
+                    DEFAULT_MAX_NURSERY
+                }
+            }
+            NurserySize::Fixed(sz) => sz,
+        };
+        conversions::raw_align_up(virtual_memory_bytes, BYTES_IN_CHUNK)
     }
 
-    /// Returns a [`NurserySize`] or [`String`] containing error. Expects nursery size to be formatted as
-    /// `<NurseryKind>:<size in bytes>`. For example, `Fixed:8192` creates a [`NurseryKind::Fixed`] nursery of size
-    /// 8192 bytes.
-    pub fn parse(s: &str) -> Result<NurserySize, String> {
-        let ns: Vec<&str> = s.split(':').collect();
-        let kind = ns[0].parse::<NurseryKind>().map_err(|_| {
-            String::from("Please specify one of \"Bounded\" or \"Fixed\" nursery type")
-        })?;
-        let value = ns[1]
-            .parse()
-            .map_err(|_| String::from("Failed to parse size"))?;
-        Ok(NurserySize::new(kind, Some(value)))
+    /// Return true if the values are valid.
+    fn validate(&self) -> bool {
+        match *self {
+            NurserySize::Bounded { min, max } => min < max,
+            NurserySize::ProportionalBounded { min, max } => min < max && max <= 1.0f64,
```

Comment:
Maybe it is useful to allow `min == max`, because we currently don't have a way to express that the nursery size shall be a particular proportion (not a range) of the heap size.
```suggestion
            NurserySize::ProportionalBounded { min, max } => 0.0f64 < min && min <= max && max <= 1.0f64
```